universal-login-wallet/src/ui/react/CreateAccount/CreateFlow.tsx

```diff
@@ -0,0 +1,78 @@
+import React, {useContext, useState} from 'react';
+import {Redirect} from 'react-router';
+import {useMutableCallback} from 'react-use-mutable';
+import {useServices} from '../../hooks';
+import {CreateAccount} from './CreateAccount';
+import {WalletModalContext} from '../../../core/entities/WalletModalContext';
+import {ETHER_NATIVE_TOKEN, GasParameters, INITIAL_GAS_PARAMETERS} from '@universal-login/commons';
+import {hideTopUpModal} from '../../../core/utils/hideTopUpModal';
+import Modal from '../Modals/Modal';
+import {ModalWrapper, TopUp, useProperty, WaitingForDeployment} from '@universal-login/react';
+import {ModalTxnSuccess} from '../Modals/ModalTxnSuccess';
+
+export function CreateFlow() {
+  const modalService = useContext(WalletModalContext);
+  const {sdk, walletService} = useServices();
+  const [gasParameters, setGasParameters] = useState<GasParameters>(INITIAL_GAS_PARAMETERS);
+  const [transactionHash, setTransactionHash] = useState<string | undefined>(undefined); // TODO: Move to wallet state
+
+  const deployWallet = useMutableCallback((name: string) => walletService.deployFutureWallet(
```

Comment:
Interesting case here. Because of the way javascript arrow functions capture variables, if I were to call `walletService.deployFutureWallet` directly in `onCreateClick` the `gasParameters` value used would be the one that was at the beginning of the async call. `useMutableCallback` here always updates the `deployWallet` function with new values each render. This way even ongoing async operation gets up-to-date data.

Reply:
Alternative to this code would be:
```
const [getGasParameters, setGasParameters] = useMutableState<GasParameters>(INITIAL_GAS_PARAMETERS);

const onCreateClick(name: string) => {
  ...
  await walletService.deployFutureWallet(name, getGasParameters());
}
```
In this case since we capture the getter for gas parameters instead of the variable itself, the value can be kept up to date.

Reply:
I'm leaning to changing the code to the second version, let me know what you think